pkg/apis/zalando.org/v1/types.go

```diff
@@ -0,0 +1,113 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+
+// ScalingSchedule describes a namespaced time based metric to be used
+// in autoscaling operations.
+// +k8s:deepcopy-gen=true
+type ScalingSchedule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ScalingScheduleSpec `json:"spec"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+
+// ClusterScalingSchedule describes a cluster scoped time based metric
+// to be used in autoscaling operations.
+// +k8s:deepcopy-gen=true
+// +kubebuilder:resource:scope=Cluster
+type ClusterScalingSchedule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ScalingScheduleSpec `json:"spec"`
+}
+
+// ScalingScheduleSpec is the spec part of the ScalingSchedule.
+// +k8s:deepcopy-gen=true
+type ScalingScheduleSpec struct {
+	// Schedules is the list of schedules for this ScalingSchedule
+	// resource. All the schedules defined here will result on the value
+	// to the same metric. New metrics require a new ScalingSchedule
+	// resource.
+	Schedules []Schedule `json:"schedules"`
+}
+
+// Schedule is the schedule details to be used inside a ScalingSchedule.
+// +k8s:deepcopy-gen=true
+type Schedule struct {
+	// Defines if the schedule is a OneTime schedule or
+	// Repeating one. If OneTime, date has to be defined. If Repeating,
+	// Period has to be defined.
+	// +kubebuilder:validation:Enum=OneTime;Repeating
+	ScheduleType string `json:"type"`
```

Comment:
nit: Maybe we want to make this into a type similar to here: https://github.com/zalando-incubator/stackset-controller/blob/6348bf62633d184eab0dbe59b0b5482d81bc460f/pkg/apis/zalando.org/v1/types.go#L173-L188

same for `ScheduleDay`

Reply:
good one, will do.